lessons/frameworks.ipynb

```diff
@@ -0,0 +1,1261 @@
+    "\n",
+    "$\\;\\;\\;\\;\\;\\;\\;m = a\\;\\log_{10} P + b$\n",
+    "\n",
+    "* The data consist of *observed magnitudes with quoted uncertainties*, such as: \n",
+    "\n",
+    "$\\;\\;\\;\\;\\;\\;\\;m^{\\rm obs} = 24.51 \\pm 0.31$ at $\\log_{10} P = \\log_{10} (13.0/{\\rm days})$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Bayesian inference\n",
+    "\n",
+    "* We compute the posterior PDF for the parameters $a$ and $b$ given the data and the assumed model: \n",
+    "\n",
+    "$\\;\\;\\;\\;\\;{\\rm Pr}(a,b|\\boldsymbol{m}^{\\rm obs},H) \\propto {\\rm Pr}(\\boldsymbol{m}^{\\rm obs}|a,b,H)\\;{\\rm Pr}(a,b|H)$\n",
+    "\n",
+    "* We can evaluate the unnormalized posterior PDF on a grid, renormalize it numerically, and then visualize and summarize the resulting 2D function."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Probabilistic graphical model\n",
+    "\n",
+    "* Let's draw a PGM for this inverse problem, imagining our way through what we would do to generate a mock dataset like the one we have from R11.\n",
+    "\n",
+    "* If we were generating mock data, then for any plausible choice of parameters $a$ and $b$ we can predict the true magnitude $m_k$ of each star given its period $P_k$, and then add noise to simulate each observed magnitude $m^{\\rm obs}_k$.\n",
+    "\n",
+    "#### Exercise:  Draw the PGM for this problem with your neighbor, and be prepared to point out its features in 5 minutes' time"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## PGM\n",
+    "\n",
+    "<img width=\"400\" src=\"graphics/pgms_cepheids.png\"/>\n",
+    "\n",
+    "> NB. The magnitude uncertainties $\\sigma^{\\rm obs}$ are given to us in the data file; we can use them as-is if we believe them. The \"true\" magnitudes $m$ are _determined_ by our power law model."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## Building an inference\n",
+    "\n",
+    "Now let's assign PDFs for each node in the PGM, and derive the unnormalized posterior PDF for $a$ and $b$.\n",
+    "\n",
+    "We'll need:\n",
+    "\n",
+    "* The sampling distribution: ${\\rm Pr}(\\boldsymbol{m}^{\\rm obs}|\\boldsymbol{m},H)$\n",
+    "\n",
+    "* The conditional PDF for the latent variables $m_k$, ${\\rm Pr}(m_k|a,b,\\log_{10}{P_k},H)$\n",
+    "\n",
+    "* A prior PDF for our parameters: ${\\rm Pr}(a,b|H)$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "## The sampling distribution ${\\rm Pr}(\\boldsymbol{m}^{\\rm obs}|\\boldsymbol{m},H)$\n",
```

Comment:
A stylistic change I've been making is to remove repeated slide headers. When showing things in slide mode, it doesn't seem especially necessary and just takes up a lot of space. When browsing the notebook in the usual way, it just looks crazy to have these repeated headers breaking up what is otherwise a cogently organized document.

Reply:
Sounds good - Ill take a pass through and follow suit.